pkg/apis/v1alpha1/awsnodetemplate.go

```diff
@@ -97,6 +100,16 @@ type AWSNodeTemplate struct {
 	Status AWSNodeTemplateStatus `json:"status,omitempty"`
 }
 
+func (a *AWSNodeTemplate) Hash() string {
+	hash, _ := hashstructure.Hash(a.Spec, hashstructure.FormatV2, &hashstructure.HashOptions{
+		SlicesAsSets:    true,
+		IgnoreZeroValue: true,
+		ZeroNil:         true,
+	})
+
+	return fmt.Sprint(hash)
```

Comment:
```suggestion
	return string(hash)
```

It think it's basically the same but probably better to not rely on `fmt` here

Reply:
Since we are converting from `uint64` to a string, the ci-test will not pass

Reply:
Oh! this is a uint64. My bad. Didn't realize it wasn't a string.